client/Components/login.jsx

```diff
@@ -65,9 +67,10 @@ const Login = props => {
       {signUp && !user && <SignUp closeModal={signUpClickHandler} setNewUser={setUser}/>}
       {!instructions && <InstructionsButton click={toggleInstructionsHandler} />}
       {instructions && <Instructions onFinish={toggleInstructionsHandler} />}
+      {user && <UserContext.Provider value={user}><Map /></UserContext.Provider>}
       {!user && <ColumnWrapper>
         <Logo />
-        <Input onFocus={onFocusInputHandler}onChange={usernameHandler}className={'sign-up-input login'} type={'text'} id={'username'} />
+        <Input onFocus={onFocusInputHandler} onChange={usernameHandler} className={'sign-up-input login'} type={'text'} id={'username'} />
         <Input onFocus={onFocusInputHandler} className={'sign-up-input mb1'} type={'password'} id={'password'} onChange={passwordHandler} />
         {validLoginInput === 'empty' && <div className="row justify-align-center"><p className="password-check">username and password cannot be empty</p></div>}
```

Comment:
The conditional rendering is getting pretty crazy in here. Try adding hash routing so you can simplify your view swapping. You can add a `<Redirect />` component to return if the user has to be logged in to view a specific page. Instead of putting these all in Login, switch between them in App